### PR TITLE
#13 Registration new users. 

### PR DESCRIPTION
**Pull request by IgorNazaryok**
#13 . Registration new users. Add  store. Methods implemented through action

image

### CODE REVIEW

File: front/src/store/modules/ReristrationUser.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  },
+  state: {
+    mailboxBusy: false
+  },
+  getters: {
+    getMailboxBusy (state) {
+      return state.mailboxBusy;
+    }
+  }
+};
```
Comment (horba) [inline front/src/store/modules/ReristrationUser.js]: please, provide correct name for store module file
http://joxi.ru/4AkqZyjtXgwXym

by https://github.com/horba/film-finity/wiki/Frontend-Namings-and-Structure

**Review by horba — CHANGES_REQUESTED** [review]

File: front/src/App.vue [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
-      <router-link to="/">Home</router-link>
-    </div>
-    <router-view />
+    <RegForm/>
   </div>
 </template>
 
 <script>
+
+import RegForm from './components/Registration/RegForm.vue';
```
Comment (horba) [inline front/src/App.vue]: use `@components`

**Review by horba — CHANGES_REQUESTED** [review]
please, read carefully about file namings and structure our wiki https://github.com/horba/film-finity/wiki/Frontend-Namings-and-Structure

File: front/src/components/Registration/RegForm.html [hunk 3]
```diff
@@ -0,0 +1,82 @@
+<div class="font">
```
Comment (horba) [inline front/src/components/Registration/RegForm.html]: font should be setted in body in common.css

File: front/src/components/Registration/RegForm.css [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  font-weight: 600;
+  padding-top: 40px;
+  align-items: flex-end;
+  text-align: center;
+  color: #292c33;
+  font-size: 18px;
+}
+a {
+  text-decoration: none;
+  color: #292c33;
```
Comment (horba) [inline front/src/components/Registration/RegForm.css]: use css variables everywhere possible, please

**Review by horba — CHANGES_REQUESTED** [review]

File: front/src/components/Registration/RegForm.html [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      </div>
+    </div>
+    <div id="created" class="blackout">
+      <div class="window">
+        <div class="text">Создать аккаунт
+          <a href="#" class="close">
+            <el-button icon="el-icon-close" circle></el-button>
+          </a>
+        </div>
+        <div class="accountCreate font">
```
Comment (horba) [inline front/src/components/Registration/RegForm.html]: account-create

File: front/src/components/Registration/RegForm.vue [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<template src="./RegForm.html"></template>
+
+<script src="./RegForm.js"></script>
+
+<style scoped src="./RegForm.css"></style>
```
Comment (horba) [inline front/src/components/Registration/RegForm.vue]: no shortcuts, please
we have article in wiki about ninja code. please, read it

File: server/Entities/Models/JobTitle.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+using System.Threading.Tasks;
+
+namespace WebAPI.Models
+{
+    public class JobTitle
+    {
+        public int JobTitleId { get; set; }
+        public string JobName { get; set; }
+
+        public ICollection<CelebrityJobTitles> CelebrityJobTitles { get; set; }
```
Comment (horba) [inline server/Entities/Models/JobTitle.cs]: it seems we have some entities for job titles for celebrities, ask @Yaronberg about it

File: server/WebAPI/Controllers/CelebrityController.cs [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+            {
+                CelebrityId = o.CelebrityId,
+                FirstName = o.FirstName,
+                LastName = o.LastName,
+                CountViews = o.CountViews,
+                ImageSource = o.ImageSource,
+                JobTitles = o.CelebrityJobTitles.Select(j => j.JobTitle).Select( j => new JobTitleDTO 
+                { JobTitleId = j.JobTitleId, JobName = j.JobName }).ToList()
+            }).ToList();
+
```
Comment (horba) [inline server/WebAPI/Controllers/CelebrityController.cs]: you can leave it but in future you need separate service for business logic. usually there is no logic in controller

Comment (horba) [reply server/WebAPI/Controllers/CelebrityController.cs]: do use automapper?

**Review by horba — CHANGES_REQUESTED** [review]

**Review by horba — CHANGES_REQUESTED** [review]
please, use components from css-framework
before making task check if some component already exists in element ui

https://element.eleme.io/#/en-US/component/dialog

Also, please, attach screens of implemented views to pull request description and in comment to task

File: front/src/components/index.js [hunk 9]
```diff
@@ -1,2 +0,0 @@
-export { default as HelloWorldOneFile } from './HelloWorldOneFile.vue';
-export { default as HelloWorld } from './hello-world/hello-world.component.vue';
```
Comment (horba) [inline front/src/components/index.js]: it is useful file
before deleting think why it was added
yes, hello-world were removed but it is place to export all components for shortcut import just only from `@components`

Comment (horba) [reply front/src/components/index.js]: https://www.freecodecamp.org/news/how-to-use-es6-modules-and-why-theyre-important-a9b20b480773/

Comment (horba) [reply front/src/components/index.js]: https://medium.com/groww-engineering/module-aliasing-in-webpack-f02fe1b91f94

**Review by horba — CHANGES_REQUESTED** [review]
index.js is a part of file infrastructure

File: server/WebAPI/Controllers/CelebrityController.cs (hunk 8 above)
Comment (IgorNazaryok) [reply server/WebAPI/Controllers/CelebrityController.cs]: Я не понял твои замечания по серверу. Этот код Ярослава. Что я должен сделать, исправить?

**Pull request closed without merging**